nova/bigvm/manager.py

```diff
@@ -324,6 +325,28 @@ def _get_providers(self, context):
             inventory = client._get_inventory(context, rp_uuid)
             rp['inventory'] = inventory['inventories']
 
+        # make sure grouping by hv_size works properly later on, even if there
+        # are marginal differences in the reported hv_size. we need to have all
+        # vmware_providers to have the same hv_size if they're in the same
+        # "bucket", e.g. they're supposed to be 3 TB HVs. To make sure the
+        # placement query for allocation-candidates works, we use the smallest
+        # size and assign it to all in the same bucket.
+        hv_size_bucket = None
+        for rp_uuid, rp in sorted(vmware_providers.items(),
+                                  key=lambda x: x[1]['hv_size']):
```

Comment:
Is there any chance these two nested lookups might ever fail?

Reply:
`x[1]` can't fail, because `.items()` returns a tuple.

`['hv_size']` can only fail, if we have a resource-provider in `vmware_providers`, that doesn't have that attribute. [This](https://github.com/sapcc/nova/blob/85b649f734eb741461facaaaf09d36bc78329ce7/nova/bigvm/manager.py#L311) is the only place where things are added to that dictionary.

Given above two points, I'd say: no.

Reply:
Alright, then go! :)